File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.*;

class Solution {
    public String solution(int[] food) {
        StringBuilder sb1 = new StringBuilder();
        StringBuilder sb2 = new StringBuilder();
        
        for (int i = 1; i < food.length; i++) {
            int count = food[i];
            
            for (int j = 0; j < count / 2; j++) {
                sb1.append(i);
                sb2.insert(0, i);
            }
        }
        return sb1.append(0).append(sb2).toString();
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    let arr = [];
    for (let i = 1; i < food.length; i++) {
        for (let j = 0; j < Math.floor(food[i] / 2); j++) {
            arr.push(i.toString());
        }
    }
    let str1 = arr.join('');
    let str2 = arr.reverse().join('');
    return str1 + '0' + str2;
}
